MetaMask/metamask-extension
# Add Petnames Name Component to Transaction Approval Screen

## **Description**

### Currently
The extension uses the SenderToRecipient React component to display the from and to address when creating most types of transaction. This is displayed in the header of most transaction confirmations:

![Image](https://github.com/MetaMask/MetaMask-planning/assets/19745142/3db7c831-58e4-4fff-a7cc-84abf594f4d7)

This internally uses a RecipientWithAddress component to render the target of the transaction.

This currently also supports names and nicknames but by using state from the AddressBookController, EnsController, and the names of the wallet accounts.

### Goal of this PR 
* use the Name component in the RecipientWithAddress component to use petnames to display the transaction to address.
* Continue to use the usePetnamesEnabled hook to only use the Name component if the petnames build feature is enabled.

### Notes

Once petnames is in main, the build feature can be removed and the legacy code removed in a separate PR.

See the Name component storybook entry for usage details.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1625
Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1626

## **Manual testing steps**
In a flask build: 
1. Click the Send button on the home screen
2. Fill out an address to send to.
3. On the confirmation screen you should see the recipient inside a name component (shaped like a pill) with either a warning or a bookmark icon, depending on whether or not there is a petname for this address-chain combination.

## **Screenshots/Recordings**



### **Before**

![Image](https://github.com/MetaMask/MetaMask-planning/assets/19745142/3db7c831-58e4-4fff-a7cc-84abf594f4d7)

### **After**

<img width="650" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/507015/2e2b6b4f-977e-46af-84f0-4389824dcab2">



## **Pre-merge author checklist**

## **Pre-merge reviewer checklist**
